File: day_trader/sources/strategy/strategies/day_trader/strategy/services/websocket/common/DTO/positions/position_update_req.py
from sources.strategy.strategies.day_trader.business_entities.day_trading_position import *
from sources.strategy.strategies.day_trader.strategy.services.websocket.common.DTO.websocket_message import *
import json

class PositionUpdateReq(WebSocketMessage):


    def __init__(self, Msg,ReqId,PosId,SharesQuantity=None,Active=None,UUID=None, TradingMode=None,
                 Depurate=None,CleanStopLoss=None,CleanTakeProfit=None,CleanEndOfDay=None):
        super(PositionUpdateReq, self).__init__(Msg)
        self.PosId=PosId
        self.SharesQuantity=SharesQuantity
        self.Active = Active
        self.UUID = UUID
        self.TradingMode = TradingMode
        self.Depurate=Depurate
        self.ReqId = ReqId
        self.CleanStopLoss=CleanStopLoss
        self.CleanTakeProfit=CleanTakeProfit
        self.CleanEndOfDay=CleanEndOfDay

    def toJSON(self):
        return json.dumps(self, default=lambda o: o.__dict__,sort_keys=True)

File: test_clients/C#/day_trader_testclient/DayTraderTestClient/DayTraderTestClient.Common/DTO/Positions/PositionUpdateReq.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DayTraderTestClient.Common.DTO.Positions
{
    public class PositionUpdateReq : WebSocketMessage
    {
        #region Public Attributes

        public string UUID { get; set; }

        public string ReqId { get; set; }

        public int PosId { get; set; }

        public int? SharesQuantity { get; set; }

        public bool? Active { get; set; }

        public bool? Depurate { get; set; }

        public string TradingMode { get; set; }

        public bool? CleanStopLoss { get; set; }

        public bool? CleanTakeProfit { get; set; }

        public bool? CleanEndOfDay { get; set; }

        #endregion
    }
}
